### ISSUE TITLE
Connecting through SOCKS mode shows connected, but no connection is routing through the VPN.

### ISSUE THREAD

**Issue by uncharted-user** [AdguardTeam/AdGuardVPNCLI#59]
### Please answer the following question for yourself before submitting an issue

- [X] I checked to make sure that this issue has not already been filed

### AdGuard VPN CLI version

v1.0.0 and v1.1.112

### Environment

- OS: Fedora Linux 40 (Workstation Edition)
- Kernel: 6.10.11-200.fc40.x86_64
- Desktop: GNOME 46.5
- Session: Wayland

### Issue Details

Steps to reproduce:

1. Open Terminal and type `curl -fsSL https://raw.githubusercontent.com/AdguardTeam/AdGuardVPNCLI/master/scripts/release/install.sh | sh -s -- -v`
2. Then login using `adguardvpn-cli login`
3. After that, connect using `adguardvpn-cli connect -l SG` → Type yes then type no and again type no
4. Then type `adguardvpn-cli config set-mode SOCKS`
5. Finally, type  `adguardvpn-cli connect -l SG` -> Check connection using `adguardvpn-cli status`

### Expected Behavior

All the connection should go through VPN.

### Actual Behavior

- After step 3 the output is "Failed to start the VPN service in the background"
- After step 5 adguard-cli will show connected but when you will type `curl --silent "https://ipinfo.io/country"` in the terminal then you will see that the country is not changed to Singapore or whatever you picked.

### Screenshots

<details><summary>Screenshot 1</summary>

</details>

**Comment by sfionov** [comment]
@uncharted-user You should specify SOCKS proxy address to apps in this case:

```
curl -x socks5://127.0.0.1:1080 --silent "https://ipinfo.io/country"
```

**Comment by uncharted-user** [comment]
> @uncharted-user You should specify SOCKS proxy address to apps in this case:
> 
> ```
> curl -x socks5://127.0.0.1:1080 --silent "https://ipinfo.io/country"
> ```

Yes, it is now showing SG. But the instructions from AdGuard did not mention anything else to modify the browser. In browser, I am still getting the same country. I want my all connection to go through VPN when connected. I couldn't find any instructions in AdGuard related to that.

**Comment by sfionov** [comment]
@uncharted-user Hello! You may use TUN mode in adguardvpn-cli to redirect all traffic, or Firefox Extension like [SwitchyOmega](https://addons.mozilla.org/en-US/firefox/addon/switchyomega/) to switch proxy on and off on the fly.

**Comment by uncharted-user** [comment]
> @uncharted-user Hello! You may use TUN mode in adguardvpn-cli to redirect all traffic, or Firefox Extension like [SwitchyOmega](https://addons.mozilla.org/en-US/firefox/addon/switchyomega/) to switch proxy on and off on the fly.

TUN mode is not working ([issue 58](https://github.com/AdguardTeam/AdGuardVPNCLI/issues/58)). Thanks for the extension.

**Comment by sfionov** [comment]
@uncharted-user In case of adguardvpn-cli, SOCKS server is running locally, so "unencrypted" part is only between your browser and adguardvpn-cli, which then use secure connection to server.

**Comment by uncharted-user** [comment]
> which then use secure connection to server.

That means anything getting out from my device is fully encrypted. Is this also applies for the macOS app? Is it feasible to incorporate a feature that would enable the Adguard-cli to route all connections through it once SOCKS is configured?

**Comment by sfionov** [comment]
There's no plan for automatically configuring SOCKS proxy in system. To redirect all traffic, TUN mode exists. So I recommend to wait for TUN mode fix for your system.

**Comment by uncharted-user** [comment]
> So I recommend to wait for TUN mode fix for your system.

No problem, I hope it will be solved soon.

**Comment by sfionov** [comment]
@uncharted-user AdGuard VPN CLI 1.2 is released with TUN mode fixes, can you please try it?

**Comment by uncharted-user** [comment]
I need a little more time to test it and get back.

**Comment by uncharted-user** [comment]
> [@uncharted-user](https://github.com/uncharted-user) AdGuard VPN CLI 1.2 is released with TUN mode fixes, can you please try it?

I tested this on Arch based distribution (EndeavourOS) and found TUN is working. Thank you for the fix. However, the speed is still terrible 👎 

![Image](https://github.com/user-attachments/assets/3c947253-829a-4f61-9682-9f406f2f1de3)
